tests/deployment/headnodes.py

```diff
@@ -40,10 +40,17 @@ def server_init():
     server.validate_state()
 
 
+@pytest.yield_fixture
```

Comment:
Hm, interesting.  If this is pretty standard, then okay.  Otherwise maybe comment it